frontend/src/metabase-types/api/mocks/dataset.ts

```diff
@@ -66,24 +67,37 @@ export const createMockPublicDatasetData = ({
     }),
   ],
   ...opts
-}: Partial<PublicDatasetData>): PublicDatasetData => ({
+}: Partial<EmbedDatasetData>): EmbedDatasetData => ({
   rows: [],
   cols,
   rows_truncated: 0,
   insights: null,
   ...opts,
 });
 
-export const createMockPublicDataset = ({
-  data = {},
-  ...opts
-}: MockDatasetOpts = {}): PublicDataset => ({
-  data: createMockPublicDatasetData(data),
-  database_id: 1,
-  row_count: 0,
-  running_time: 1000,
-  ...opts,
-});
+export type MockEmbedDatasetOpts = Omit<Partial<EmbedDataset>, "data"> & {
+  data?: Partial<EmbedDatasetData>;
+};
+
+export const createMockEmbedDataset = (
+  opts: MockEmbedDatasetOpts,
+): EmbedDataset => {
+  if ("data" in opts) {
```

Comment:
This change is due to the fact that `EmbedDataset` can either be `SuccessEmbedDataset` or `ErrorEmbedDataset`, and they have different shapes.